modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandlerTest.java

```diff
@@ -0,0 +1,132 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import org.apache.ignite.internal.configuration.testframework.ConfigurationExtension;
+import org.apache.ignite.internal.configuration.testframework.InjectConfiguration;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.configuration.storage.DataStorageConfiguration;
+import org.apache.ignite.internal.storage.ReadResult;
+import org.apache.ignite.internal.storage.RowId;
+import org.apache.ignite.internal.storage.index.IndexStorage;
+import org.apache.ignite.internal.table.distributed.raft.PartitionDataStorage;
+import org.apache.ignite.internal.util.Cursor;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * For {@link StorageUpdateHandler} testing.
+ */
+@ExtendWith(ConfigurationExtension.class)
+public class StorageUpdateHandlerTest {
+    private static final int PARTITION_ID = 0;
+
+    @InjectConfiguration
+    private DataStorageConfiguration dataStorageConfig;
+
+    private final HybridClock clock = new HybridClockImpl();
+
+    @Test
+    void testBuildIndex() {
+        PartitionDataStorage partitionStorage = mock(PartitionDataStorage.class);
+
+        TableSchemaAwareIndexStorage indexStorage = createIndexStorage();
+
+        UUID indexId = UUID.randomUUID();
+
+        TableIndexStoragesSupplier indexes = mock(TableIndexStoragesSupplier.class);
+
+        when(indexes.get()).thenReturn(Map.of(indexId, indexStorage));
+
+        StorageUpdateHandler storageUpdateHandler = createStorageUpdateHandler(partitionStorage, indexes);
+
+        RowId rowId0 = new RowId(PARTITION_ID, UUID.randomUUID());
+        RowId rowId1 = new RowId(PARTITION_ID, UUID.randomUUID());
+
+        List<BinaryRow> rowVersions0 = asList(mock(BinaryRow.class), null);
+        List<BinaryRow> rowVersions1 = asList(mock(BinaryRow.class), null);
+
+        setRowVersions(partitionStorage, Map.of(rowId0.uuid(), rowVersions0, rowId1.uuid(), rowVersions1));
+
+        storageUpdateHandler.buildIndex(indexId, List.of(rowId0.uuid(), rowId1.uuid()), false);
```

Comment:
It's interesting that you pass rowIds in potentially incorrect order. You shouldn't do it, because it's not realistic and it may spontaneously stop working in the future if we add some assertion somewhere.

Reply:
I am passing in a list of `RowId` and expect that we will only execute code relative to order of that list, not the order of the `RowId`.
I don't check order `RowId`, I assume that the receiver sends the `RowId` in the correct order.